idea/src/org/jetbrains/kotlin/idea/inspections/RedundantIfInspection.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2010-2016 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.*
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElementVisitor
+import org.jetbrains.kotlin.KtNodeTypes
+import org.jetbrains.kotlin.psi.*
+
+class RedundantIfInspection : AbstractKotlinInspection(), CleanupLocalInspectionTool {
+
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean, session: LocalInspectionToolSession): PsiElementVisitor {
+        return object : KtVisitorVoid() {
+            override fun visitIfExpression(expression: KtIfExpression) {
+                super.visitIfExpression(expression)
+
+                if (expression.condition != null && isRedundant(expression)) {
+                    holder.registerProblem(expression,
+                                           "Redundant 'if' statement",
+                                           ProblemHighlightType.LIKE_UNUSED_SYMBOL,
+                                           RemoveRedundantIf)
+                }
+            }
+        }
+    }
+
+    private fun isRedundant(expression: KtIfExpression): Boolean {
+        val thenReturn = getReturnedExpression(expression.then) ?: return false
+        val elseReturn = getReturnedExpression(expression.`else`) ?: return false
+
+        if (KtPsiUtil.isTrueConstant(thenReturn) && !KtPsiUtil.isTrueConstant(elseReturn)) {
```

Comment:
I find it somewhat confusing that the KtNodeTypes.BOOLEAN_CONSTANT is separate from the value check. It would be clearer to add the corresponding `isFalseConstant()` method to `KtPsiUtil` and to remove the node type check from `getReturnedExpression()`.